papermill/execute.py

```diff
@@ -182,9 +182,11 @@ def execute_notebook(notebook,
     nb.metadata.papermill['exception'] = any(
         [cell.metadata.papermill.get('exception') for cell in nb.cells])
 
-    # Write final Notebook to disk.
-    write_ipynb(nb, output)
-    raise_for_execution_errors(nb, output)
+    if output:
+        # Write final Notebook to disk.
+        write_ipynb(nb, output)
+        raise_for_execution_errors(nb, output)
```

Comment:
with this only executed when we write the notebook, what happens if there is an exception and the notebook isn't written to a file?

Reply:
@betatim Thanks, perhaps I have a wrong indent with `raise_for_execution_errors(nb, output)`